internals/src/Tizen.Peripheral/Tizen.Peripheral/Uart.cs

```diff
@@ -0,0 +1,354 @@
+        AutoRtsCts
+    }
+
+    /// <summary>
+    /// Enumeration for software flow control.
+    /// </summary>
+    public enum SoftwareFlowControl
+    {
+        /// <summary>No software flow control.</summary>
+        None = 0,
+
+        /// <summary>XON/XOFF software flow control.</summary>
+        XonXoff,
+    }
+
+    /// <summary>
+    /// The class allows applications to communicate via UART platform's bus.
+    /// </summary>
+    /// <privilege>http://tizen.org/privilege/peripheralio</privilege>
+    public class Uart : IDisposable
+    {
+        private BaudRate _baudRate;
+        private ByteSize _byteSize;
+        private Parity _parity;
+        private StopBits _stopBits;
+        private HardwareFlowControl _hardwareFlowControl;
+        private SoftwareFlowControl _softwareFlowControl;
+        private bool _disposed = false;
+
+        /// <summary>
+        /// Native handle to I2c.
+        /// </summary>
+        private IntPtr _handle;
+
+        /// <summary>
+        /// Opens the UART slave device.
+        /// </summary>
+        /// <param name="port">The UART port number that the slave device is connected.</param>
+        public Uart(int port)
+        {
+            var ret = NativeUart.Open(port, out IntPtr handle);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// Closes the UART slave device.
+        /// </summary>
+        ~Uart()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Closes the UART slave device.
+        /// </summary>
+        public void Close() => Dispose();
+
+        /// <summary>
+        /// Disposes Uart object.
+        /// </summary>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Disposes Uart object.
+        /// </summary>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            NativeUart.Close(_handle);
+            _disposed = true;
+        }
+
+        /// <summary>
+        /// Reads the bytes data from the UART slave device.
+        /// </summary>
+        /// <param name="buffer">The output byte array.</param>
+        /// <param name="offset">The offset in buffer at which to write the bytes.</param>
+        /// <param name="count">The number of bytes to read.</param>
+        public void Read(byte[] buffer, int offset, int count)
+        {
+            if (count > buffer.Length - offset)
+                throw new Exception("Can not read more bytes than the buffer can hold.");
+            byte[] tmpBuffer = new byte[count];
+            var length = Convert.ToUInt32(count);
+            var ret = NativeUart.Read(_handle, tmpBuffer, length);
```

Comment:
Let's see it, it is source code of native API
``` c
int peripheral_interface_uart_read(peripheral_uart_h uart, uint8_t *buf, uint32_t length)
{
    /* the return type has to allow negative values (for error codes), so we can't use the upper half of unsigned values */
    int ret = read(uart->fd, (void *)buf, length > INT_MAX ? INT_MAX : length);
    CHECK_ERROR(ret < 0);

    return ret;
}
```
it return read length, you should not throw exception when ret was not zero, 
And Please test before submit